Support array namespaces for module nesting

jskrzypek opened the pull request:
### Summary
This PR addresses two issues, #15 and #34, by
 * replacing `deep-assign` and `clone` dependencies with `lodash.merge` and `lodash.clonedeep`
 * refactoring the namespace resolution utils to support array namespaces and to be more DRY
 * refactoring `stripSlashes()` in `utils` to support arrays and use `lodash.trim` instead of the regex

marshallswain commented:
@jskrzypek this is beautiful.  Looks like a minor release?  Nothing breaking in here, right?

jskrzypek commented:
@marshallswain 
Nope, I did not change any tests to make them pass. I also changed `getNameFromConfig()` to be `getNameFromExplicit()` to better match the `nameStyles` value, but I updated all of the references to the function, so nothing breaks.

daffl commented:
@marshallswain I can make the release if you add me to npm.

marshallswain commented:
@daffl that would be great.  I added you.  Thanks!

daffl commented:
Released as v0.8.0. Thank you for the PR!
